<launch>

  <arg name="gui" default="true"/>
  <arg name="camera" default="rarm_hand_camera" />
  <arg name="INPUT_IMAGE" default="/$(arg camera)/color/image_raw" />
  <arg name="INPUT_CAMERA_INFO" default="/$(arg camera)/color/camera_info" />

  <group ns="checkerdetector" clear_params="true">
    <!-- display 0: (default) not display / 1: display found corner -->
    <param name="display" type="int" value="1"/>
    <!-- maxboard -1: (default) infinite search board / 1<=: limit found number of boards -->
    <param name="maxboard" type="int" value="1"/>
    <param name="max_error_threshold" value="0.3" />
    <!-- set frame_id if you want to use specific frame_id -->
    <!-- param name="frame_id" type="string" value="stereo_link"/-->
    <param name="rect0_size_x" type="double" value="0.03"/>
    <param name="rect0_size_y" type="double" value="0.03"/>
    <param name="grid0_size_x" type="int" value="6"/>
    <param name="grid0_size_y" type="int" value="5"/>
    <param name="type0" type="string" value="6x5chessboard"/>

    <param name="adaptive_thresh" value="true" />
    <param name="use_P"              value="false" /> <!-- if you use rectified image, use_P should be true -->
    <param name="invert_color"       value="false" /> <!-- invert white and black -->
    <param name="message_throttle"   value="1" />
    <param name="queue_size"         value="1" />
    <param name="publish_queue_size" value="1" />
    <param name="axis_size"   value="0.05" /> <!-- set axis_size as [m] -->
    <param name="circle_size" value="6" />    <!-- set circle_size as [pixel] -->

    <node name="detector_node"
          pkg="checkerboard_detector" type="checkerboard_detector"
          respawn="false" output="screen">
      <remap from="camera_info" to="$(arg INPUT_CAMERA_INFO)"/>
      <remap from="image" to="$(arg INPUT_IMAGE)"/>
    </node>

    <node name="objectdetection_tf_publisher"
          pkg="checkerboard_detector"
          type="objectdetection_tf_publisher.py" >
      <param name="use_simple_tf" value="true" />
    </node>

  </group>

  <group if="$(arg gui)" >
    <node name="$(anon rviz)"
          pkg="rviz" type="rviz"
          args="-d $(find vzense_demo)/launch/config/hand_camera_calib.rviz" >
    </node>
  </group>

</launch>
